### DISCUSSION TITLE
[Feature Request] Vehicle Gear Mod Hook

### DISCUSSION THREAD

**Discussion by postmvn** [SmartlyDressedGames/Unturned-3.x-Community#4658] [Ideas]
We have **gears** now, which is great. But they do feel empty and boring as there is no sounds or effects played when gears are changed.
I believe it should be possible to implement something like gear mod hook for vehicles to trigger events on gear change. This would make modded vehicles much more immersive and realistic.

It could look something like that:

**On Gear Changed**
Trigger an event each time gear changed

**On Gear Changed Up** 
Trigger an event each time gear changed up 

**On Gear Changed Down** 
Trigger an event each time gear changed down

**On Forward Gear Changed**
Trigger an event only when forward gears changed (e.g from 1 to 2, from 5 to 6 or from 3 to 1)

**On Forward Gear**
Trigger an event when gear is changed to one of forward ones (could be used to mimic D gear on automatic gearbox)

**On Reverse Gear**
Trigger an event when gear is set to reverse (would be very useful for proper reverse lights etc)

**On Neutral Gear**
Trigger an event when neutral gear is selected

I included those that should be mostly used, but there could be more.
Not sure how hard it is to sync such system but even if it will be local it shouldn't be a problem.

**Comment by SDGNelson** [comment]
I agree this is a good idea. I would like to have some vanilla effects related to engines and gears as part of vehicle improvements but this would provide modders a lot of flexibility in the meantime.
